src/balena-host-envvars.service

```diff
@@ -4,7 +4,7 @@ Description=balena-host-envvars
 [Service]
 Type=oneshot
 EnvironmentFile=/etc/docker.env
-ExecStart=/usr/sbin/configure-balena-host-envvars.sh
+ExecStart=/bin/sh -c 'while ! /usr/sbin/configure-balena-host-envvars.sh; do sleep 3s; done'
```

Comment:
I'd rather see these done via `Restart=on-failure` which is available for oneshot services since systemd 2.44 (I think the image has a new enough systemd?)